lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -874,26 +884,31 @@ sub _add_job {
     # add parents
     for my $parent ($job->parents->all) {
         my ($parent_job, $dependency_type) = ($parent->parent, $parent->dependency);
-        my $parent_job_id = _add_job($visited, $nodes, $edges, $cluster, $cluster_by_job, $parent_job) or next;
+        my $parent_job_id
+          = _add_job($visited, $nodes, $edges, $cluster, $cluster_by_job, $parent_job, $as_child_of, $preferred_depth)
```

Comment:
if that wraps over multiple lines you know you have too many parameters :) Can you refactor that?